workflow/common/util.go

```diff
@@ -290,9 +314,58 @@ func ProcessArgs(tmpl *wfv1.Template, args wfv1.Arguments, globalParams, localPa
 	}
 	tmpl.Inputs.Artifacts = newInputArtifacts
 
+	//copy passthrough parameters from Inputs to Arguments
+	if tmpl.Steps != nil {
+		for i, stepList := range tmpl.Steps {
+			for j, step := range stepList {
+				updatedParameters, err := handlePassthroughParameters(tmpl.Inputs, step.Arguments.Parameters)
+				if err != nil {
+					return nil, err
+				}
+				tmpl.Steps[i][j].Arguments.Parameters = updatedParameters
+			}
+		}
+	}
+	if tmpl.DAG != nil {
+		for i, task := range tmpl.DAG.Tasks {
+			updatedParameters, err := handlePassthroughParameters(tmpl.Inputs, task.Arguments.Parameters)
+			if err != nil {
+				return nil, err
+			}
+			tmpl.DAG.Tasks[i].Arguments.Parameters = updatedParameters
+		}
+	}
+
 	return substituteParams(tmpl, globalParams, localParams)
 }
 
+// handlePassthroughParameters returns parameter argument list where items matching PassthroughRegexp are added from input parameters
+func handlePassthroughParameters(inputs wfv1.Inputs, argumentParameters []wfv1.Parameter) ([]wfv1.Parameter, error) {
+	updatedArgumentParameters := make([]wfv1.Parameter, 0)
+
+	// if the input still contains passthrough parameters (eg from FakeArguments.GetParametersByRegexp)
+	// then we can't resolve any passthrough parameters here
+	inputsContainsRegexp := inputs.DoesContainRegexpParameters()
+
+	for _, parameter := range argumentParameters {
+		if parameter.PassthroughRegexp != "" && !inputsContainsRegexp {
+
+			inputParams, err := inputs.GetParametersByRegexp(parameter.PassthroughRegexp)
+			if err != nil {
+				return nil, errors.Errorf(errors.CodeBadRequest, "regexp %s could not be processed", parameter.PassthroughRegexp)
+			}
+
+			for _, inputParam := range inputParams {
+				newParameter := wfv1.Parameter{Name: inputParam.Name, Value: inputParam.Value, Regexp: inputParam.Regexp}
```

Comment:
We're copying parameters from arguments to inputs, so I think this `Regexp: inputParam.Regexp` is not necessary.

Reply:
If the top-level arguments aren't available (eg we're validating a WorkflowTemplate) then the regex needs to flow through ...

Reply:
I think we don’t have to care WorkflowTemplate so much in this PR. That is the next step.

Reply:
Ok, I'll fix this up once #1508 is merged.

Reply:
#1508 has been merged.